lib/internal/streams/destroy.js

```diff
@@ -375,31 +379,20 @@ function emitErrorCloseLegacy(stream, err) {
   process.nextTick(emitCloseLegacy, stream);
 }
 
-function isDestroyed(stream) {
-  return stream.destroyed || stream[kDestroyed];
-}
-
-function isReadable(stream) {
-  return stream.readable && !stream.readableEnded && !isDestroyed(stream);
-}
-
-function isWritable(stream) {
-  return stream.writable && !stream.writableEnded && !isDestroyed(stream);
-}
-
 // Normalize destroy for legacy.
 function destroyer(stream, err) {
   if (isDestroyed(stream)) {
     return;
   }
 
-  if (!err && (isReadable(stream) || isWritable(stream))) {
+  if (!err && !isFinished(stream)) {
     err = new AbortError();
   }
 
   // TODO: Remove isRequest branches.
-  if (typeof stream[kDestroy] === 'function') {
-    stream[kDestroy](err);
+  if (isServerRequest(stream)) {
+    stream.socket = null;
```

Comment:
Why do we need to prevent the socket from destroying in this case? Is there any chance to move this logic into IncomingMessage.prototype._destroy to decouple destroy logic a bit?

Reply:
So that the response is not destroyed. No it’s not possible to decouple to due breaking reasons. Unfortunately this is the only way.